src/main/webapp/js/query.js

```diff
@@ -421,6 +424,158 @@ async function fetchFeedback() {
+            if (submitButtonEl != null) {
+                submitButtonEl.remove();
+            }
+            modal.style.display = "none";
+        }
+        // When the user clicks anywhere outside of the modal, close it
+        window.onclick = event => {
+            if (event.target == modal) {
+                let restaurantContainerEl = document.getElementById("restaurant-name-container");
+                restaurantContainerEl.remove();
+                let submitButtonEl = document.getElementById("submit-button");
+                if (submitButtonEl != null) {
+                    submitButtonEl.remove();
+                }
+                modal.style.display = "none";
+            }
+        }
+        feedbackButton = document.createElement('button');
+        feedbackButton.className = 'btn1 feedback';
+        feedbackButton.innerText = "Submit Feedback";
+        feedbackButton.addEventListener('click', () => {
+            let restaurantNameEl = createRestaurantElement(search.name);
+            formEl.appendChild(restaurantNameEl);
+            modal.style.display = "block";
+        });
+        newCardBody.appendChild(feedbackButton);
+    }
+    let searchButton = document.createElement('button');
+    searchButton.className = 'btn1 search';
+    searchButton.innerText = "Search with These Parameters Again";
+    newCardBody.appendChild(searchButton);
+    searchButton.addEventListener('click', () => {
+        reroll()});
+    return newCardBody;
+}
+
+//Function to append restaurant name to modal form to force it to follow through to feedback
+function createRestaurantElement(restaurantName) {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    let userEl = document.createElement('input');
+    userEl.className = "input--style-2";
+    userEl.type = "text";
+    userEl.id = "user-id";
+    userEl.name = "user-id";
+    userEl.value = userID;
+    userEl.hidden = true;
+
+    let inputGroupEl = document.createElement('div');
+    inputGroupEl.className = "input-group";
+    inputGroupEl.id = "restaurant-name-container";
+    let inputContainer = document.createElement('input');
+    inputContainer.className = "input--style-2";
+    inputContainer.type = "text";
+    inputContainer.id = "restaurant-name-fill";
+    inputContainer.name = "restaurant-name-fill";
+    inputContainer.value = restaurantName;
+    inputContainer.innerText = restaurantName;
+    inputGroupEl.appendChild(inputContainer);
+    inputGroupEl.appendChild(userEl);
+
+    let submitEl = document.createElement('input');
+    submitEl.type = "submit";
+    submitEl.id = "submit-button";
+    inputGroupEl.appendChild(submitEl);
+    return inputGroupEl;
+}
+
+async function fetchFeedback() {
+    let userID = 0;
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    let response = await fetch(`/feedback?user=${userID}`, {
```

Comment:
i think i left this comment in #34 but we should make sure to chain promises together, since this await breaks the chain and waits for the promise to resolve before continuing the execution of the chain